### PR TITLE
feat:(SRE-portal) Initialize plugin

### PR DESCRIPTION
**Pull request by mimotej**
This PR intializes ~~SRE-portal~~ Web terminal plugin and adds dummy terminal to it which is not connected to any backend. ~~plugin can be found on `/sre-portal` address.~~
Edit: Reworked to contain only webterminal component which can be added to actual instance of app
Example:
image

### CODE REVIEW

File: packages/app/package.json [hunk 1]
```diff
@@ -46,13 +46,14 @@
     "@backstage/plugin-techdocs-react": "^1.0.5",
     "@backstage/plugin-user-settings": "^0.5.0",
     "@backstage/theme": "^0.2.16",
+    "@internal/backstage-plugin-rhacm": "*",
+    "@internal/plugin-sre-portal": "^0.1.0",
```
Comment (tumido) [inline packages/app/package.json:50]: Let's not version this for now and please rename it to `@internal/backstage-plugin-web-terminal`

File: plugins/sre-portal/package.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{
+  "name": "@internal/plugin-sre-portal",
```
Comment (tumido) [inline plugins/sre-portal/package.json:2]: see above

File: plugins/sre-portal/src/components/TerminalComponent/TerminalComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  });
+  return (
+    <>
+      <div className={classes.root}>
+        <h1 className={classes.header}>SRE portal</h1>
+        <div className={classes.termcontainer}>
+          <div className={classes.term} id="terminal" />
+        </div>
+      </div>
+    </>
```
Comment (tumido) [inline plugins/sre-portal/src/components/TerminalComponent/TerminalComponent.tsx:49]: This view should be build within the instance only. This plugin should provide a web terminal only.


We don't aim to reimplement prometheus plugin or techdocs plugin in this one, we want to integrate them. Such integration builds on atomic parts - plugins. It should be build within the instance as a new view/page utilizing the existing extensions. This extension is one integrated member not the one integrating it all together. We don't want to introduce plugin bundles like and creating too opinionated plugins.

**Review by tumido — CHANGES_REQUESTED** [review]

File: packages/app/package.json (hunk 1 above)
Comment (mimotej) [reply packages/app/package.json]: Resolved

File: plugins/sre-portal/src/components/TerminalComponent/TerminalComponent.tsx (hunk 3 above)
Comment (mimotej) [reply plugins/sre-portal/src/components/TerminalComponent/TerminalComponent.tsx]: :+1: Didn't realize that we want each part separately and combine them in the instance... Reworked

File: plugins/web-terminal/src/plugin.test.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { srePortalPlugin } from './plugin';
+
+describe('sre-portal', () => {
```
Comment (tumido) [inline plugins/web-terminal/src/plugin.test.ts:3]: `web-terminal`

**Review by tumido — CHANGES_REQUESTED** [review]

File: plugins/web-terminal/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# sre-portal
+
+Welcome to the sre-portal plugin!
```
[the remark this answers is not in the document]
Comment (mimotej) [reply plugins/web-terminal/README.md]: Resolved

File: plugins/web-terminal/dev/index.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { createDevApp } from '@backstage/dev-utils';
+import { webTerminalPlugin, WebTerminal } from '../src/plugin';
+
+createDevApp()
+  .registerPlugin(webTerminalPlugin)
+  .addPage({
+    element: <WebTerminal />,
+    title: 'Root Page',
+    path: '/sre-portal',
```
[the remark this answers is not in the document]
Comment (mimotej) [reply plugins/web-terminal/dev/index.tsx]: Resolved

File: plugins/web-terminal/src/components/TerminalComponent/TerminalComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    const terminalElement = document.getElementById('terminal');
+    if (terminalElement) {
+      terminal.open(terminalElement);
+      terminal.write('$ ');
+    }
+  });
+  return (
+    <div className={classes.termcontainer}>
+      <div className={classes.term} id="terminal" />
+    </div>
```
[the remark this answers is not in the document]
Comment (mimotej) [reply plugins/web-terminal/src/components/TerminalComponent/TerminalComponent.tsx]: Leftover from previous iteration fixed :+1:

File: plugins/web-terminal/src/plugin.test.ts (hunk 4 above)
Comment (mimotej) [reply plugins/web-terminal/src/plugin.test.ts]: Resolved

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**